roxie/ccd/ccdfile.cpp

```diff
@@ -2051,31 +2051,35 @@ class CRoxieFileCache : implements IRoxieFileCache, implements ICopyFileProgress
         closePending[remote] = false;
         IArrayOf<ILazyFileIO> goers;
         {
-            CriticalBlock b(crit);
-            HashIterator h(files);
-            ForEach(h)
+            if (files.ordinality() > maxFilesOpen[remote] || maxFileAgeNS[remote] != (unsigned __int64) -1)
             {
-                ILazyFileIO * match = files.mapToValue(&h.query());
-                if (match->isAliveAndLink())
+                CriticalBlock b(crit);   // Does it matter that we test files.ordinality() outside the critblock?
```

Comment:
I think the test should really be inside the CriticalBlock, although in practice I can't see it causing any problems except on weird architectures that do not update 32bit values atomically.

Reply:
I'm not sure there is any significant performance impact, so may as well code it correctly.